### DISCUSSION TITLE
[v4] - How to delete with relationship

### DISCUSSION THREAD

**Discussion by kempsteven** [adonisjs/core#1857] [Help] [answered]
Soo I have a `Delivery` Model and has a `hasMany` relationship with `SelectedDriver` Model
and the `SelectedDriver` Model also have a `hasMany` relationship with Invoice Model and other models.

```js
class Delivery extends Model {
  selected_driver() {
    return this.hasMany('App/Models/SelectedDriver')
  }
}
```

```js
class SelectedDriver extends Model {
  user() {
    return this.belongsTo('App/Models/User')
  }

  delivery_orders() {
    return this.hasMany('App/Models/DeliveryOrder')
  }

  stocks() {
    return this.hasMany('App/Models/Stock')
  }

  estimated_revenue() {
    return this.hasOne('App/Models/EstimatedRevenue')
  }

  invoices() {
    return this.hasMany('App/Models/Invoice')
  }
}
```

so what I want to ask is what is the best and easiest way to delete all of them when I delete A Delivery??
When I delete a Delivery, the Selected Drivers and Invoices should be deleted as well.
Thanks in Advance for the help!

**Comment by thetutlage** [comment, marked as answer]
This is something that will be well done within the database using the `on delete cascade` option. If you are unaware of how this option works in SQL, then I recommend first reading a bit about and then you can use AdonisJS migrations to set the `on delete cascade` when creating database tables
